programs/invariant/src/util.rs

```diff
@@ -185,4 +195,149 @@ mod test {
         }
         Ok(())
     }
+
+    #[test]
+    fn test_cross_tick() -> Result<()> {
+        //conception
+        let mut test_pool = Pool {
+            fee_growth_global_x: FeeGrowth::from_integer(9),
+            fee_growth_global_y: FeeGrowth::from_integer(4),
+            liquidity: Liquidity::from_integer(2),
+            last_timestamp: 10,
+            start_timestamp: 0,
+            seconds_per_liquidity_global: FixedPoint::from_integer(8),
+            current_tick_index: 2,
+            ..Default::default()
+        };
+
+        let mut test_tick = Tick {
+            index: 3,
+            seconds_per_liquidity_outside: FixedPoint { v: (3) },
+            liquidity_change: Liquidity::from_integer(1),
+            fee_growth_outside_x: FeeGrowth::from_integer(11),
+            fee_growth_outside_y: FeeGrowth::from_integer(1),
+            seconds_outside: 5,
+
+            ..Default::default()
+        };
+        //without underfolw
+        {
+            //When pool.current_tick_index is higher then tick.index
+            {
+                //test_pool
+                test_pool.fee_growth_global_x = FeeGrowth { v: (45) };
+                test_pool.fee_growth_global_y = FeeGrowth { v: (35) };
+                test_pool.liquidity = Liquidity::from_integer(4);
+                test_pool.last_timestamp = 15;
+                test_pool.start_timestamp = 4;
+                test_pool.seconds_per_liquidity_global = FixedPoint::from_integer(11);
+                test_pool.current_tick_index = 7;
+                //test_tick
+                test_tick.fee_growth_outside_x = FeeGrowth { v: (30) };
+                test_tick.fee_growth_outside_y = FeeGrowth { v: (25) };
+                test_tick.index = 3;
+                test_tick.seconds_outside = 5;
+                test_tick.seconds_per_liquidity_outside = FixedPoint { v: (3) };
+                test_tick.liquidity_change = Liquidity::from_integer(1);
+                let mut f_t_t = RefCell::new(test_tick);
+                let mut t = f_t_t.borrow_mut();
+                cross_tick(&mut t, &mut test_pool, 18446744073709);
+                let result = t.borrow().fee_growth_outside_x;
+                assert_eq!(result, FeeGrowth { v: (15) });
```

Comment:
This pattern looks ugly:
```
let result = ...
assert_eq!(...)
```
Compare entire structures (similar as in test_swap_step function)